src/client/mod.rs

```diff
@@ -140,31 +164,54 @@ impl InfluxDbClient {
 
         let client = if let Some(_) = any_value.downcast_ref::<InfluxDbReadQuery>() {
             let read_query = query.get();
-            let encoded: String = form_urlencoded::Serializer::new(String::new())
-                .append_pair("db", self.database_name())
-                .append_pair("q", &read_query)
-                .finish();
-            let http_query_string = format!(
-                "{url}/query?{encoded}",
-                url = self.database_url(),
-                encoded = encoded
-            );
+   
+            let mut parameters = vec![
+                ("db", self.database_name()),
+                ("q", &read_query),
+            ];
+            
+            if let Some(auth) = self.auth() {
+                parameters.push(("u", auth.username.as_str()));
+                parameters.push(("p", auth.password.as_str()));
+            }
+
+            let url = match Url::parse_with_params(format!("{url}/query", url = self.database_url()).as_str(), parameters) {
+                Ok(url) => url,
+                Err(err) => {
+                    let error = InfluxDbError::InvalidQueryError {
+                        error: format!("{}", err),
+                    };
+                    return Box::new(future::err::<String, InfluxDbError>(error));
+                }
+            };
             if read_query.contains("SELECT") || read_query.contains("SHOW") {
-                Client::new().get(http_query_string.as_str())
+                Client::new().get(url)
             } else {
-                Client::new().post(http_query_string.as_str())
+                Client::new().post(url)
             }
         } else if let Some(write_query) = any_value.downcast_ref::<InfluxDbWriteQuery>() {
+            let precision_modfier = write_query.get_precision_modifier();
+            let mut parameters = vec![
```

Comment:
I agree, that is is a little bit hacky. Upon some discussion with @msrd0 we came up with a possibly cleaner solution:

There could be a struct 
```
struct InfluxDbParams {
    db: &str,
    precision: &str,
    auth: &Option< InfluxDbAuthentication>
}
```

which has an `Into<Vec<(&str, &str)>>` implementation. 

What do you think about this approach?

Reply:
Isn't the precision normally set in context of a single measurement? Mixed measurements would not be possible this way. We should on the other hand rename the function to get_precision as it now returns not the modifier but the precision.
The `Into<Vec<(&str, &str)>>` idea sounds good.

Reply:
You're right, that code should have been 

```
struct InfluxDbParams {
    db: &str,
    q: &str,
    auth: &Option< InfluxDbAuthentication>
}
```, I mixed that up.

Reply:
I am not sure about that `q` in the struct. It is only present for read queries. But we could omit that and have `auth` and `db` in that struct and then add `q` and `precision` after thet in their context.